lib/entice/logic/MapInstance.ex

```diff
@@ -0,0 +1,78 @@
+defmodule Entice.Logic.MapInstance do
+  alias Entice.Entity
+  alias Entice.Logic.MapInstance
+
+
+  defstruct(
+    players: [],
+    npcs: [],
+    map: nil)
+
+  def join(entity, player_entity) do
+    Entity.call_behaviour(entity, MapInstance.Behaviour, {:map_instance_entity_join, player_entity})
+  end
+
+  def leave(entity, player_entity) do
+    Entity.call_behaviour(entity, MapInstance.Behaviour, {:map_instance_entity_leave, player_entity})
+  end
+
+  def register(entity, map, players=[], npc_info=[]),
```

Comment:
if we accept lists of thing as input here, lets document what we expect to find in these lists